### ISSUE TITLE
Error to download remote images having long URL

### ISSUE THREAD

**Issue by kovax** [sphinx-doc/sphinx#4688]
Subject: Build was triggered manually

https://readthedocs.org/projects/cristalise-kernel/builds/6818449/

### Problem

OSError: [Errno 36] File name too long: '/home/docs/checkouts/readthedocs.org/user_builds/cristalise-kernel/checkouts/latest/_build/doctrees-epub/images/https/www.planttext.com/plantuml/img/VLAnJiCm4Dtz5IUpWLpHjIfH4LqwWK2T48CRcyGeYIFx0aAb_yx5IUmKYKxHtRrtxzsBDnwbmwwk6Df2OwaT71ayIInX_IC4FDjAzzMyjqDT_-rjRIrD4xedDW66qNsX6moJD12lfL5ADs-LVaUH9Hac3nwTp14UAkdzgB2UkbdMsmFnWiuq1RI37algm5lfI0-M74YRvwMf55ppcK5O0UTiKloN4041VYP1Kp4QvxUB1J7ZOIkWF2b85g-60I_29zqP6NmPfw7dRAiT6kqP6oA8oFxo8_qlDFa9RiO8dBK8Ty19pU0t6GMuskztM-Wwe_hYyV2XAKvZNP_KR6KK2elFIUFoHjPwLdxAgjDfoooIYzPKmF54d5Auo7842fmyS5AUymSm6aplmVilwnJHDhfHzAn_0G00'
The full traceback has been saved in /tmp/sphinx-err-ghQvDs.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!

#### Procedure to reproduce the problem
```
<Paste your command-line here which cause the problem>
```

#### Error logs / results
```
<Paste your error log here>
```
- <public link of unexpected result if you have>

#### Expected results
<Describe what to actually do>

### Reproducible project / your project
- <link to your project, or attach zipped small project sample>

### Environment info
- OS: <Unix/Linux/Mac/Win/other with version>
- Python version:
- Sphinx version:
- <Extra tools e.g.: Browser, tex or something else>

**Comment by tk0miya** [comment]
Sorry for inconvenience.

>OSError: [Errno 36] File name too long: 

The error is caused at downloading remote images. It seems github wiki uses hash codes for its URL:
https://github.com/cristal-ise/kernel/wiki/Lookup

**Comment by kovax** [comment]
This is not a mistake, I use plantuml to draw diagrams to document the software. Check this URL:

https://www.planttext.com/plantuml/img/VLAnJiCm4Dtz5IUpWLpHjIfH4LqwWK2T48CRcyGeYIFx0aAb_yx5IUmKYKxHtRrtxzsBDnwbmwwk6Df2OwaT71ayIInX_IC4FDjAzzMyjqDT_-rjRIrD4xedDW66qNsX6moJD12lfL5ADs-LVaUH9Hac3nwTp14UAkdzgB2UkbdMsmFnWiuq1RI37algm5lfI0-M74YRvwMf55ppcK5O0UTiKloN4041VYP1Kp4QvxUB1J7ZOIkWF2b85g-60I_29zqP6NmPfw7dRAiT6kqP6oA8oFxo8_qlDFa9RiO8dBK8Ty19pU0t6GMuskztM-Wwe_hYyV2XAKvZNP_KR6KK2elFIUFoHjPwLdxAgjDfoooIYzPKmF54d5Auo7842fmyS5AUymSm6aplmVilwnJHDhfHzAn_0G00

The diagram is encoded in the hash, which is actually a textual description of the class diagram, and this RESTAPI of planttext returns the rendered image to be show in the wiki.

It would be great if you could support such images, I think it is a very convenient way to enhance the documentation.

**Comment by tk0miya** [comment]
Yes, I know this is a bug of Sphinx. I'll work for it later.

**Comment by kovax** [comment]
Currently this error stops the entire build, and if I go to the project link I see nothing:

http://cristalise-kernel.readthedocs.io/en/latest/

I would expect to get most of my docs imported even there was an error. Is it correct? Do I miss some other steps? Would it be possible to generate the docs even if there was an error?

**Comment by tk0miya** [comment]
In Sphinx-1.7, the very long image URL causes Sphinx crashed.
So please pin the version of Sphinx to 1.6.x or more older version.

**Comment by tk0miya** [comment]
Fixed by #4747.
Thank you for reporting.
